### PR TITLE
feat(salesforce): OAuth flow, sync users & deactivate users

### PR DESCRIPTION
**Pull request by athielba**
## Description

1. OAuth flow
2. Sync users
3. Schedule user sync
4. Deactivate user
## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests to cover the new feature or fixes.

---------

## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/salesforce/src/inngest/middlewares/rate-limit-middleware.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            if (!(error instanceof SalesforceError)) {
+              return;
+            }
+            // TODO: remove this once discussed with Guillaume & Alex
+            // if (error.response?.status === 403) {
+            //   const data = await error.response.text();
+            //   const errors = JSON.parse(data) as {
+            //     errorCode: string;
+            //     message: string;
+            //   }[];
```
Comment (llanicm) [inline apps/salesforce/src/inngest/middlewares/rate-limit-middleware.ts:25]: Did you have this discussion with @proquig and @elbalexandre?

**Review by llanicm — COMMENTED** [review]
One small comment base on a comment in the code, other than that it LGTM

File: apps/salesforce/src/inngest/middlewares/rate-limit-middleware.ts (hunk 1 above)
Comment (athielba) [reply apps/salesforce/src/inngest/middlewares/rate-limit-middleware.ts]: Since it is uncommon case, i removed the commented code & added comment

**Review by llanicm — APPROVED** [review]
LGTM

**Pull request merged**